### PR TITLE
Review opset1 split for shape inference aspects

### PR DESCRIPTION
**Pull request by praasz**
### Details:
 - Extend testing shape inference for dimension propagation and label propagation
 - Add evaluate lower, upper, and label with tests
 - Use template shape inference in evaluate

### Tickets:
 - 90620
 - 90717
 - 90754
 - 90791
 - 90828
 - 90865
 - 90902
 - 90939

### CODE REVIEW

File: src/core/tests/type_prop/squeeze.cpp [hunk 1]
```diff
@@ -261,7 +261,7 @@ TEST_P(SqueezeBoundTest, propagate_label_and_dynamic_value) {
     const auto unsqueeze = std::make_shared<op::v0::Unsqueeze>(gather, axis_1);
     const auto squeeze = std::make_shared<op::v0::Squeeze>(unsqueeze, axis);
 
-    const auto bc = std::make_shared<op::v1::Broadcast>(param, squeeze);
+    const auto bc = std::make_shared<op::v3::Broadcast>(param, squeeze);
```
Comment (jane-intel) [inline src/core/tests/type_prop/squeeze.cpp:264]: smth wrong with `v1::Broadcast` or just upgrading to use latest versions?

**Review by jane-intel — APPROVED** [review]

File: src/core/tests/type_prop/squeeze.cpp (hunk 1 above)
Comment (praasz) [reply src/core/tests/type_prop/squeeze.cpp:264]: Just upgrade to latest one, and @mitruska improve in this version some checks.

**Comment by ilya-lavrenov** [comment]
Looks like Android build is broken after this change https://dev.azure.com/openvinoci/dldt/_build/results?buildId=423745&view=logs&j=92ba3447-fcaa-530d-6438-b829fe1e7d1d&t=b0a91e75-8666-5718-e63e-df215ecb48f9&l=2804

**Comment by praasz** [comment]
> Looks like Android build is broken after this change https://dev.azure.com/openvinoci/dldt/_build/results?buildId=423745&view=logs&j=92ba3447-fcaa-530d-6438-b829fe1e7d1d&t=b0a91e75-8666-5718-e63e-df215ecb48f9&l=2804

Unused variables removed in PR #13894

**Pull request merged**